src/components/association-dialog.js

```diff
@@ -0,0 +1,336 @@
+import { selectStyles } from '@brightspace-ui/core/components/inputs/input-select-styles.js';
+import { TccServiceFactory } from '../services/tccServiceFactory';
+
+const DEFAULT_ASSOCIATION = {
+	Department: {
+		OrgUnitId: '-1',
+		Name: ''
+	},
+	Prefix: '',
+	Suffix: '',
+	Role: {
+		Id: -1,
+		Name: ''
+	}
+};
+
+class TccAssociationDialog extends BaseMixin(LitElement) {
+
+	static get properties() {
+		return {
+			tccService: {
+				type: Object
+			},
+			association: {
+				type: Object
+			},
+			associationDialogOpened: {
+				attribute: 'opened',
+				reflect: true,
+				type: Boolean
+			},
+			isNewAssociation: {
+				type: Boolean
+			},
+			roles: {
+				attribute: 'roles',
+				type: Array
+			},
+			departments: {
+				attribute: 'departments',
+				type: Array
+			},
+			associationForm: {
+				type: Object
+			}
+		};
+	}
+
+	static get styles() {
+		const associationDialogStyles = css`
+			:host {
+				display: inline-block;
+			}
+
+			:host([hidden]) {
+				display: none;
+			}
+
+			.association_form {
+			}
+
+			.association_form__input_group {
+				margin: 6px;
+				display: flex;
+				flex-direction: column;
+			}
+
+			.association_form__button_group {
+				margin: 6px;
+			}
+
+			.association_form__button {
+				width: 102px;
+			}
+		`;
+		return [
+			associationDialogStyles,
+			inputStyles,
+			inputLabelStyles,
+			selectStyles
+		];
+	}
+
+	constructor() {
+		super();
+
+		this.tccService = TccServiceFactory.getTccService();
+
+		this.associationDialogOpened = false;
+		this.isNewAssociation = false;
+		this.association = {...DEFAULT_ASSOCIATION};
+		this.invalidFlags = {
+			Department: false,
+			Prefix: false,
+			Suffix: false,
+			Role: false
+		};
+	}
+
+	async open(associationToEdit) {
+		if (!associationToEdit) {
+			this._formReset();
+		} else {
+			this.association = associationToEdit;
+		}
+
+		this.associationDialogOpened = true;
+		await this.requestUpdate();
+
+		if (!this.associationForm) {
+			this.associationForm = this._formLoad();
+		}
+	}
+
+	async _close() {
+		this.associationDialogOpened = false;
+	}
+
+	_formReset() {
+		this.isNewAssociation = true;
+		this.association = {...DEFAULT_ASSOCIATION};
+
+		Object.keys(this.invalidFlags).map(
+			flag => this.invalidFlags[flag] = false
+		);
+
+		if (this.associationForm) {
+			this.associationForm.DepartmentSelect.selectedIndex = 0;
+			this.associationForm.RoleSelect.selectedIndex = 0;
+			this.associationForm.PrefixInput.value = '';
+			this.associationForm.SuffixInput.value = '';
+		}
+	}
+
+	_formLoad() {
+		return {
+			DepartmentSelect: this.shadowRoot.querySelector('#association-department'),
+			PrefixInput: this.shadowRoot.querySelector('#association-prefix'),
+			SuffixInput: this.shadowRoot.querySelector('#association-suffix'),
+			RoleSelect: this.shadowRoot.querySelector('#association-role')
+		};
+	}
+
+	_formIsValid() {
+		return Object.values(this.invalidFlags).filter(flag => flag === true).length === 0;
```

Comment:
Does something like this work?
```suggestion
		return !Object.values(this.invalidFlags).includes(flag => flag ===true);
```

Reply:
Not exactly but I can do something with a similar intention; I do see that this conditional can be simplified.